### PR TITLE
Convert useShowPassword hook to TypeScript file

### PR DESCRIPTION
**Pull request by wariofdz**
**Changed:**

- Transform useShowPassword hook from JS file to TS file.

### CODE REVIEW

File: packages/frontend/web-ui/src/common/hooks/useShowPassword.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+export const useShowPassword = (initial: boolean) => {
+  const [showPassword, setShowPassword] = useState<boolean>(initial);
+
+  const handleClickShowPassword = () => {
+    setShowPassword((show: boolean) => !show);
+  };
+
+  const handleMouseDownPassword = (
+    event: React.FormEvent<HTMLInputElement>,
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/common/hooks/useShowPassword.ts:11]: We don't really care which element is being targeted (btw, it's a button, not an input), so `Event<HTMLElement>` is way more convenient

**Review by notaphplover — CHANGES_REQUESTED** [review]
Looks greeat, just a minor comment

File: packages/frontend/web-ui/src/common/hooks/useShowPassword.ts (hunk 1 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/common/hooks/useShowPassword.ts:11]: I've seen another type to named this element, you can use it for any event: **_e: React.SyntheticEvent_**

File: packages/frontend/web-ui/src/common/hooks/useShowPassword.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { useState } from 'react';
+
+export const useShowPassword = (initial: boolean) => {
+  const [showPassword, setShowPassword] = useState<boolean>(initial);
+
+  const handleClickShowPassword = () => {
+    setShowPassword((show: boolean) => !show);
+  };
+
+  const handleMouseDownPassword = (event: React.FormEvent<HTMLElement>) => {
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/common/hooks/useShowPassword.ts:10]:
```suggestion
  const handleMouseDownPassword = (event: React.MouseEvent<HTMLElement>) => {
```

File: packages/frontend/web-ui/src/common/hooks/useShowPassword.ts (hunk 1 above)
Comment (notaphplover) [reply packages/frontend/web-ui/src/common/hooks/useShowPassword.ts]: `React.MouseEvent` is probably more convenient then

**Review by notaphplover — CHANGES_REQUESTED** [review]
Let's use a mouse event for the mouse down event handler

File: packages/frontend/web-ui/src/common/hooks/useShowPassword.ts (hunk 2 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/common/hooks/useShowPassword.ts]: Done!

File: packages/frontend/web-ui/src/common/hooks/useShowPassword.ts (hunk 1 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/common/hooks/useShowPassword.ts]: Done!

**Review by notaphplover — APPROVED** [review]

**Pull request merged**